bindings.go

```diff
@@ -388,3 +390,141 @@ func (n Node) PrevNamedSibling() *Node {
 func (n Node) Edit(i EditInput) {
 	C.ts_node_edit(&n.c, i.c())
 }
+
+// QueryErrorType - value that indicates the type of QueryError.
+type QueryErrorType int
+const (
+	QueryErrorNone QueryErrorType = iota
+	QueryErrorSyntax
+	QueryErrorNodeType
+	QueryErrorField
+	QueryErrorCapture
+)
+
+// QueryError - if there is an error in the query,
+// then the Offset argument will be set to the byte offset of the error,
+// and the Type argument will be set to a value that indicates the type of error.
+type QueryError struct {
+	Offset uint32
+	Type QueryErrorType
+}
+func (qe *QueryError) Error() string {
+	switch qe.Type {
+	case QueryErrorNone:
+		return ""
+
+	case QueryErrorSyntax:
+		return fmt.Sprintf("syntax error (offset: %d)", qe.Offset)
+
+	case QueryErrorNodeType:
+		return fmt.Sprintf("node type error (offset: %d)", qe.Offset)
+
+	case QueryErrorField:
+		return fmt.Sprintf("field error (offset: %d)", qe.Offset)
+
+	case QueryErrorCapture:
+		return fmt.Sprintf("capture error (offset: %d)", qe.Offset)
+
+	default:
+		return fmt.Sprintf("unknown error (offset: %d)", qe.Offset)
+	}
+}
+
+// Query API
+type Query struct { c *C.TSQuery }
+
+// NewQuery creates a query by specifying a string containing one or more patterns.
+func NewQuery(pattern []byte, lang *Language) (*Query, *QueryError) {
```

Comment:
The return types here are not exactly user-friendly, specifically the error. Is there a particular reason to return a nil `*QueryError` instead of a nil `error` interface?

Reply:
`QueryError` implements an `error` interface, so it's a regular `error`, but returning an error means  we'll loose information about an error type and offset (unless you know to cast it). In other words, we if have this information (concrete type) I don't see the reason not to return it.